### ISSUE TITLE
Problem with `get-block-info?` in epoch 3.0

### ISSUE THREAD

**Issue by obycode** [stacks-network/stacks-core#5310]
**Describe the bug**
~~There is some problem computing the `block-height` in a Clarity 2 contract deployed in epoch 3.0.~~
The problem is with the usage of `get-block-info?` here. See follow up comments.

**Steps To Reproduce**
See for example this contract:
```clarity
(define-read-only (what-time-is-it)
  (print {
    block: (- block-height u1),
    time: (get-block-info? time (- block-height u1))
  })
)
```
deployed at [`ST34CYSTX9853GG3TDY4FQZ9FM6F386MT0FHM0DMS.time-check`](https://explorer.hiro.so/txid/0x184413d3acdfdbdc599c2e01b5f092475ea7a3cfde217ec8e528a5e0476aa4c2?chain=testnet).

**Expected behavior**
~~`block-height` should return the current tenure height. It is incorrect. It seems to be incrementing with each new tenure, but the starting height is incorrect.~~ See followup.

**Comment by obycode** [comment]
`block-height` is returning the tenure height in epoch 3, but then `get-block-info?` is using that as a block height to look up the block info.

**Comment by obycode** [comment]
I think what we're going to want to do here is to make `get-block-info?` expect a tenure height, not a Stacks block height. This is a consensus breaking change.

**Comment by obycode** [comment]
Fixed in #5321.
